### PR TITLE
Commiting the 1st layer of injecting DB connection to application

### PR DESCRIPTION
**Pull request by vantihovich**

### CODE REVIEW

File: postgres/users.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	return &UsersProvider{db: *db}
+}
+
+func (u *UsersProvider) FindUserByEmailAndPassword(email, password string) (*User, error) {
+	us := User{Email: email, Password: password}
+	row := u.db.QueryRow(`Select id from users where username=$1 and password=$2;`, email, password)
+	if err := row.Scan(&us.ID); err != nil {
+		return nil, err
+	}
+	log.WithFields(log.Fields{}).Info("The query executed successfully")
```
Comment (apaliavy) [inline postgres/users.go:27]: why do you need log.Fields{} here?

File: postgres/users.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	Password string `json:"password"`
+}
+
+func NewUsersProvider(db *DB) *UsersProvider {
+	return &UsersProvider{db: *db}
+}
+
+func (u *UsersProvider) FindUserByEmailAndPassword(email, password string) (*User, error) {
+	us := User{Email: email, Password: password}
+	row := u.db.QueryRow(`Select id from users where username=$1 and password=$2;`, email, password)
```
Comment (apaliavy) [inline postgres/users.go:23]: Could you please slightly change your SQL dialect? For example: 
```
SELECT id FROM users WHERE username=$1 AND password=$2
```

File: postgres/users.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+func NewUsersProvider(db *DB) *UsersProvider {
+	return &UsersProvider{db: *db}
+}
+
+func (u *UsersProvider) FindUserByEmailAndPassword(email, password string) (*User, error) {
+	us := User{Email: email, Password: password}
```
Comment (apaliavy) [inline postgres/users.go:22]: Have you gofmt'ed your code?

File: postgres/users.go (hunk 1 above)
Comment (vantihovich) [reply postgres/users.go:27]: It was added just for developing needs. Should I add any logging within this commit?

File: postgres/users.go (hunk 3 above)
Comment (vantihovich) [reply postgres/users.go:22]: No, will do in next commits

File: postgres/users.go (hunk 2 above)
Comment (vantihovich) [reply postgres/users.go:23]: Done

**Review by apaliavy — APPROVED** [review]

**Pull request merged**